vscode-dotnet-runtime-library/src/Acquisition/GenericDistroSDKProvider.ts

```diff
@@ -0,0 +1,183 @@
+
+    public async installDotnet(fullySpecifiedVersion : string): Promise<string>
+    {
+        const supportStatus = await this.getDotnetVersionSupportStatus(fullySpecifiedVersion);
+        if(supportStatus === DotnetDistroSupportStatus.Microsoft)
+        {
+            const preinstallCommands = this.myVersionPackages()[this.preinstallCommandKey];
+            const preparationResult = (await this.commandRunner.execute(preinstallCommands))[0];
+        }
+
+        let command = this.myDistroCommands()[this.installCommandKey];
+        const sdkPackage = this.myDotnetVersionPackages(fullySpecifiedVersion)[this.sdkKey];
+        command = command.replace('{0}', sdkPackage);
+        const commandResult = await this.commandRunner.execute(command);
+
+        return commandResult[0];
+    }
+
+    public async getInstalledGlobalDotnetPathIfExists() : Promise<string | null>
+    {
+        const commandResult = await this.commandRunner.execute(this.myDistroCommands()[this.currentInstallPathCommandKey]);
+        return commandResult[0];
+    }
+
+    public async dotnetPackageExistsOnSystem(fullySpecifiedDotnetVersion : string) : Promise<boolean>
+    {
+        let command = this.myDistroCommands()[this.packageLookupCommandKey];
+        const sdkPackage = this.myDotnetVersionPackages(this.JsonDotnetVersion(fullySpecifiedDotnetVersion))[this.sdkKey];
+        command = command.replace('{0}', sdkPackage);
+        const commandResult = await this.commandRunner.execute(command);
+
+        const noPackageResult = 'no packages found';
+        return commandResult[0].includes(noPackageResult);
+    }
+
+    public getExpectedDotnetDistroFeedInstallationDirectory(): Promise<string>
+    {
+        return this.myDistroCommands()[this.expectedDistroFeedInstallDirKey];
+    }
+
+    public getExpectedDotnetMicrosoftFeedInstallationDirectory(): Promise<string>
+    {
+        return this.myDistroCommands()[this.expectedMicrosoftFeedInstallDirKey];
+    }
+
+    public async upgradeDotnet(versionToUpgrade : string): Promise<string>
+    {
+        let command = this.myDistroCommands()[this.updateCommandKey];
+        const sdkPackage = this.myDotnetVersionPackages(versionToUpgrade)[this.sdkKey];
+        command = command.replace('{0}', sdkPackage);
+        const commandResult = await this.commandRunner.execute(command);
+
+        return commandResult[0];
+    }
+
+    public async uninstallDotnet(versionToUninstall : string): Promise<string>
+    {
+        let command = this.myDistroCommands()[this.uninstallCommandKey];
+        const sdkPackage = this.myDotnetVersionPackages(versionToUninstall)[this.sdkKey];
+        command = command.replace('{0}', sdkPackage);
+        const commandResult = await this.commandRunner.execute(command);
+
+        return commandResult[0];
+    }
+
+    public async getInstalledDotnetSDKVersions(): Promise<string[]>
+    {
+        const command = this.myDistroCommands()[this.installedSDKVersionsCommandKey];
```

Comment:
I like this strategy of keeping the distro commands in a json file it can be easily extended in the future without having to dig through the code 👏

Reply:
Thank you! :) It is an idea involving multiple folks, Rich in particular, so I cant take all of the credit!